pkg/controller/project/project_namespace_control.go

```diff
@@ -0,0 +1,204 @@
+	// ReconcileProject implements the control logic for Project namespace creation, update, and deletion.
+	// If an implementation returns a non-nil error, the invocation will be retried using a rate-limited strategy.
+	// Implementors should sink any errors that they do not wish to trigger a retry, and they may feel free to
+	// exit exceptionally at any point provided they wish the update to be re-run at a later point in time.
+	ReconcileProjectNamespace(namespace *corev1.Namespace, key string) error
+}
+
+// NewDefaultProjectNamespaceControl returns a new instance of the default implementation ControlInterface that
+// implements the documented semantics for Projects. updater is the UpdaterInterface used
+// to update the status of Projects. You should use an instance returned from NewDefaultProjectNamespaceControl() for any
+// scenario other than testing.
+func NewDefaultProjectNamespaceControl(k8sGardenClient kubernetes.Client, kubeInformerFactory kubeinformers.SharedInformerFactory, namespaceLister kubecorev1listers.NamespaceLister, projectLister gardenlisters.ProjectLister) ControlInterfaceProjectNamespace {
+	return &defaultProjectNamespaceControl{k8sGardenClient, kubeInformerFactory, namespaceLister, projectLister}
+}
+
+type defaultProjectNamespaceControl struct {
+	k8sGardenClient kubernetes.Client
+	kubeInformers   kubeinformers.SharedInformerFactory
+	namespaceLister kubecorev1listers.NamespaceLister
+	projectLister   gardenlisters.ProjectLister
+}
+
+func (c *defaultProjectNamespaceControl) ReconcileProjectNamespace(obj *corev1.Namespace, key string) error {
+	// We only reconcile namespaces which were marked to be used as a project.
+	if role, ok := obj.Labels[common.GardenRole]; !ok || role != common.GardenRoleProject {
+		return nil
+	}
+
+	var (
+		namespace              = obj.DeepCopy()
+		projectNamespaceLogger = logger.NewFieldLogger(logger.Logger, "project_namespace", namespace.Name)
+
+		projectName  string
+		projectOwner = rbacv1.Subject{
+			APIGroup: rbacv1.GroupName,
+			Kind:     rbacv1.UserKind,
+			Name:     "unknown",
+		}
+		projectPurpose     *string
+		projectDescription *string
+
+		project *gardenv1beta1.Project
+		err     error
+
+		projectKind         = "Project"
+		mustCreateProject   bool
+		mustUpdateOwnerRefs = true
+	)
+
+	// Determine project name for the given namespace object.
+	project, err = common.ProjectForNamespace(c.k8sGardenClient, namespace)
+	if err != nil {
+		return err
+	}
+	if project == nil {
+		mustCreateProject = true
+		projectName = common.ProjectNameForNamespace(namespace)
+		if owner, ok := namespace.Annotations[common.ProjectOwner]; ok {
+			projectOwner.Name = owner
+		}
+		if purpose, ok := namespace.Annotations[common.ProjectPurpose]; ok {
+			projectPurpose = &purpose
+		}
+		if description, ok := namespace.Annotations[common.ProjectDescription]; ok {
+			projectDescription = &description
+		}
+	} else {
+		mustCreateProject = false
```

Comment:
If project already exists then you should stop the reconcile. 
The owner references and so should be updated by the `Project` controller.

Reply:
OK, I initially thought of just creating the `Project` object once for the existing `Namespace`s, but true, for real backward-compatibility it should be reconciled properly.
However, the project-namespace-controller will be removed in further releases as we want the only way to create projects to be via the `Project` resource itself.